tests/ui/async-await/return-type-notation/issue-110963-early.stderr

```diff
@@ -29,9 +35,16 @@ LL | |         if !hc.check().await {
 LL | |             log_health_check_failure().await;
 LL | |         }
 LL | |     });
-   | |______^
+   | |______^ one type is more general than the other
+   |
+   = note: expected trait `Send`
+              found trait `for<'a> Send`
```

Comment:
Why do we end up reporting this?

Reply:
I'm not sure :) seems like something with our lowering isn't quite right. I've seen something similar before, I think.